### PR TITLE
Feature/read mapped content

### PR DESCRIPTION
**Pull request by khatton-ft**

### CODE REVIEW

File: content/draft_content_rw.go [hunk 1]
```diff
@@ -16,23 +18,69 @@ const (
...
+}
+
+func NewDraftContentRWService(endpoint string, mapper DraftContentMapper) DraftContentRW {
+	return &draftContentRW{pacExternalService{endpoint, &http.Client{}}, mapper}
+}
+
+func (rw *draftContentRW) Read(ctx context.Context, contentUUID string) (io.ReadCloser, error) {
+	tid, err := tidutils.GetTransactionIDFromContext(ctx)
+	if err != nil {
+		log.Warn("context contains no transaction id")
```
[the remark this answers is not in the document]
Comment (khatton-ft) [reply content/draft_content_rw.go]: Discard this error, it isn't really useful.

File: content/draft_content_rw.go [hunk 2]
```diff
@@ -16,23 +18,69 @@ const (
...
-	return &draftContentRW{pacExternalService{endpoint, &http.Client{}}}
+func (rw *draftContentRW) readNativeContent(tid string, contentUUID string) (*http.Response, error) {
+	readLog := log.WithField(tidutils.TransactionIDKey, tid).WithField("uuid", contentUUID)
+
+	req, err := http.NewRequest("GET", fmt.Sprintf(rwURLPattern, rw.endpoint, contentUUID), nil)
+	if err != nil {
+		readLog.WithError(err).Error("Error in creating the HTTP read request from content RW")
+		return nil, err
+	}
+	req.Header.Set(tidutils.TransactionIDHeader, tid)
```
[the remark this answers is not in the document]
Comment (khatton-ft) [reply content/draft_content_rw.go]: Found another one too, in the write path ...

File: content/handler.go [hunk 3]
```diff
@@ -27,21 +27,35 @@ func (h *Handler) ReadContent(w http.ResponseWriter, r *http.Request) {
...
+		if err == ErrDraftNotFound {
+			readLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", uuid)
+			readLog.Warn("Draft not found in PAC, trying UPP")
+			uppResp, err := h.uppContentAPI.Get(ctx, uuid)
+			if err != nil {
+				readLog.WithError(err).Error("Error in calling Content API")
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			if uppResp.StatusCode == http.StatusOK || uppResp.StatusCode == http.StatusNotFound || uppResp.StatusCode == http.StatusBadRequest {
```
[the remark this answers is not in the document]
Comment (khatton-ft) [reply content/handler.go]: :scream: :+1:

File: content/httpClient.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+func newHttpRequest(ctx context.Context, method string, url string, payload io.Reader) (req *http.Request, err error) {
+	req, err = http.NewRequest(method, url, payload)
+	if err == nil {
+		tid, tiderr := tidutils.GetTransactionIDFromContext(ctx)
+		if tiderr == nil {
+			req.Header.Set(tidutils.TransactionIDHeader, tid)
+		}
+
+		req.Header.Set("User-Agent", "PAC-draft-content-api/" + strings.Replace(buildinfo.GetBuildInfo().Version, " ", "-", -1))
```
Comment (peteclark-ft) [inline content/httpClient.go]: Interesting... This will mean changes elsewhere in PAC if this is the most compliant value for the header

**Pull request merged**